azure-pipelines.yml

```diff
@@ -46,63 +53,94 @@ jobs:
-          echo "##vso[task.setvariable variable=PKG_CONFIG_PATH;]$MONOPREFIX/lib/pkgconfig:$MONOPREFIX/share/pkgconfig:$PKG_CONFIG_PATH"
-          echo "##vso[task.setvariable variable=PATH;]$MONOPREFIX/bin:$PATH"
-        displayName: Use Mono $(MONO_VERSION)
-        failOnStderr: false
+          MonoVersion: $(MonoVersion)
       - script: |
           echo $PATH
           chmod 755 ./build.sh
-          ./build.sh --configuration Release --verbosity Verbose --target All --publish-all --archive
+          ./build.sh --configuration Release --verbosity Verbose --target CI --publish-all --archive
         displayName: 'Build'
       - template: ./.pipelines/artifacts.yml
         parameters:
           Artifacts: $(Artifacts)
 
-  - job: Linux
+  - job: macOS_tests
     pool:
-      vmImage: "Ubuntu-18.04"
+      vmImage: "macOS-latest"
     dependsOn: GitVersion
-    variables:
-      MONO_VERSION: 6.10.0
+    strategy:
+      matrix:
+        ${{ each project in parameters.testProjects }}:
+          ${{ project }}:
+            TEST_PROJECT: ${{ project }}
     steps:
       - template: ./.pipelines/init.yml
         parameters:
           Verbosity: $(Verbosity)
           CakeVersion: $(CakeVersion)
           DotNetVersion: $(DotNetVersion)
           NuGetVersion: $(NuGetVersion)
+          MonoVersion: $(MonoVersion)
       - script: |
-          sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF
-          sudo apt install apt-transport-https ca-certificates
-          echo "deb https://download.mono-project.com/repo/ubuntu stable-bionic/snapshots/$MONO_VERSION.105 main" | sudo tee /etc/apt/sources.list.d/mono-official-stable.list
-          sudo apt update
-          sudo apt install mono-devel
-          sudo apt install msbuild
-        displayName: Use Mono $(MONO_VERSION)
+          echo $PATH
+          chmod 755 ./build.sh
+          ./build.sh --configuration Debug --verbosity Verbose --target Test --test-project="$TEST_PROJECT"
+        displayName: 'Build'
+      - template: ./.pipelines/logs.yml
+        parameters:
+          Artifacts: $(Artifacts)
+
+  - job: Linux
+    pool:
+      vmImage: "ubuntu-latest"
+    dependsOn: GitVersion
+    steps:
+      - template: ./.pipelines/init.yml
+        parameters:
+          Verbosity: $(Verbosity)
+          CakeVersion: $(CakeVersion)
+          DotNetVersion: $(DotNetVersion)
+          NuGetVersion: $(NuGetVersion)
+          MonoVersion: $(MonoVersion)
       - script: |
           chmod 755 ./build.sh
-          ./build.sh --configuration Release --verbosity Verbose --target All --publish-all --archive
+          ./build.sh --configuration Release --verbosity Verbose --target CI --publish-all --archive
         displayName: 'Build'
       - template: ./.pipelines/artifacts.yml
         parameters:
           Artifacts: $(Artifacts)
 
+  - job: Linux_tests
+    pool:
+      vmImage: "ubuntu-latest"
```

Comment:
I don't think we should use ubuntu-latest -at the moment it is still `ubuntu-18.04` but maybe one day it will change to `20.04`. i think it's better to pin to `ubuntu-18.04` for now to have deterministic CI builds

Reply:
fixed